doc: update iso installation page with new cluster cidr and dhcp settings (#705)
harvester/docs: docs/install/harvester-configuration.md

  vip: 10.10.0.19
  vip_hw_addr: 52:54:00:ec:0e:0b
  vip_mode: dhcp

>   cluster_pod_cidr: 172.16.0.0/16

Reviewer: I'd suggest we use default values here

Reviewer 2: This also reminds me below topic:

The CIDR looks to have three reserved IPv4 ranges for private network: `10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16`; other CIDR may encounter risks. @ihcsim Is this checked on Harvester-installer? thanks.

https://en.wikipedia.org/wiki/Reserved_IP_addresses

Author: @w13915984028 No, we don't have any such checks on the installer. We only make sure they are well-formed CIDRs. It will be nice if these checks can be delegated to RKE2, then RKE2 can tell us exactly what is/isn't acceptable. But AFAICT, [RKE2 also doesn't do this](https://github.com/k3s-io/k3s/blob/646e3135bc9f3afd1f18a5eb13d0a9ffb7f965a1/pkg/cli/server/server.go#L318-L329).

It might not be a bad idea to do this check in the installer, but also feel might not be necessary at this point. WDYT?

Reviewer 2: We may consider to add a suggestion in this document PR, recommend users to set subnets from the above 3 private networks.

  vip_hw_addr: 52:54:00:ec:0e:0b
```

### `install.cluster_pod_cidr`

**Definition**: CIDR of the Harvester pods.

Use this field to override the default pod CIDR of 10.52.0.0/16.

**Example**:

```yaml
install:
  cluster_pod_cidr: 172.16.0.0/16
```

### `install.cluster_service_cidr`

**Definition**: CIDR of the Harvester services.

Use this field to override the default service CIDR of 10.53.0.0/16.

> If this CIDR is changed, the cluster DNS IP must be within this range. See the `install.cluster_dns` field.

Reviewer 3: ```suggestion
:::info important

If you change this CIDR, you must ensure that the cluster DNS IP (`install.cluster_dns`) is within this range.

:::
```

**Example**:

```yaml
install:
  cluster_service_cidr: 172.22.0.0/16
```

### `install.cluster_dns`

**Definition**: IP of the Harvester DNS service.

Use this field to override the default DNS service IP of 10.53.0.10.

> This IP must be within the range defined by the `cluster_service_cidr` field.

Reviewer 3: ```suggestion
:::info important

This IP must be within the range defined by the `cluster_service_cidr` field.

:::
```
